packages/jest-util/src/Console.js

```diff
@@ -48,8 +49,10 @@ export default class CustomConsole extends Console {
   }
 
   assert(...args: Array<any>) {
-    if (args[0]) {
-      this._log('assert', format(...args.slice(1)));
+    try {
+      assert(!!args[0], ...args.slice(1));
+    } catch (error) {
+      this._log('assert', error.toString());
```

Comment:
https://nodejs.org/api/console.html#console_console_assert_value_message_args
>Note: The console.assert() method is implemented differently in Node.js than the console.assert() method available in browsers.

>Specifically, in browsers, calling console.assert() with a falsy assertion will cause the message to be printed to the console without interrupting execution of subsequent code. In Node.js, however, a falsy assertion will cause an AssertionError to be thrown.

Should we respect that? We'd have to pass environment, or detect jsdom/node otherwise. Personally I'd like `console.assert` errors to always be caught, like in browsers.

Reply:
Yes, I agree with you about it, because otherwise, the assertion error would fail the test.